src/main/java/edu/cornell/kfs/concur/web/struts/ConcurManageAccessTokenAction.java

```diff
@@ -0,0 +1,80 @@
+package edu.cornell.kfs.concur.web.struts;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.struts.action.ActionForm;
+import org.apache.struts.action.ActionForward;
+import org.apache.struts.action.ActionMapping;
+import org.kuali.kfs.kns.question.ConfirmationQuestion;
+import org.kuali.kfs.kns.web.struts.action.KualiAction;
+import org.kuali.kfs.krad.util.GlobalVariables;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.rice.core.api.config.property.ConfigurationService;
+
+import edu.cornell.kfs.concur.ConcurConstants;
+import edu.cornell.kfs.concur.ConcurKeyConstants;
+import edu.cornell.kfs.concur.service.ConcurAccessTokenService;
+
+@SuppressWarnings("deprecation")
+public class ConcurManageAccessTokenAction extends KualiAction {
+
+    public ActionForward start(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) throws Exception {
+        return mapping.findForward(KFSConstants.MAPPING_BASIC);
+    }
+
+    public ActionForward replaceToken(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) throws Exception {
+        Object question = request.getParameter(KFSConstants.QUESTION_INST_ATTRIBUTE_NAME);
+        boolean revokeOldToken;
+        
+        if (ObjectUtils.isNull(question)) {
+            if (getConcurAccessTokenService().currentAccessTokenExists()) {

```

Comment:
Do we want to give the user the option to request a new access token without revoking the old one? How is the old one going to be revoked in that case? We are leaving an access token available that we are no longer controlling.